src/pasta/fields/fq.rs

```diff
@@ -796,11 +773,36 @@ fn test_inv() {
     assert_eq!(inv, INV);
 }
 
+#[test]
+fn test_rescue() {
+    // NB: TWO_INV is standing in as a "random" field element
+    assert_eq!(
+        Fq::TWO_INV
+            .pow_vartime(&[Fq::RESCUE_ALPHA, 0, 0, 0])
+            .pow_vartime(&Fq::RESCUE_INVALPHA),
+        Fq::TWO_INV
+    );
+}
+
+#[test]
+fn test_sqrt() {
+    // NB: TWO_INV is standing in as a "random" field element
+    let v = (Fq::TWO_INV).square().sqrt().unwrap();
+    assert!(v == Fq::TWO_INV || (-v) == Fq::TWO_INV);
+}
+
+#[test]
+fn test_deterministic_sqrt() {
+    // NB: TWO_INV is standing in as a "random" field element
+    let v = (Fq::TWO_INV).square().deterministic_sqrt().unwrap();
+    assert!(v == Fq::TWO_INV || (-v) == Fq::TWO_INV);
```

Comment:
Check that this is deterministic (by asserting a specific sign).

Reply:
We have other tests which ensure that this thing behaves deterministically. (Also, we're going to be removing that function anyway since we ended up not needing it.)

Reply:
Strictly speaking the other sqrt function is deterministic too; it's just that this is a square root function which a circuit can verify produced a deterministic outcome.